extensions/amp-story/1.0/amp-story.js

```diff
@@ -1835,6 +1856,32 @@ export class AmpStory extends AMP.BaseElement {
         containsMediaElementWithAudio || hasStoryAudio);
   }
 
+  /**
+   * Checks for the presence of a sidebar. If a sidebar does exist, then an icon
+   * permitting for the opening/closing of the sidebar is shown.
+   * @private
+   */
+  initializeSidebar_() {
+    this.sidebar_ = this.element.querySelector('amp-sidebar');
+    if (this.sidebar_) {
+      this.storeService_.dispatch(Action.TOGGLE_HAS_SIDEBAR,
+          !!this.sidebar_);
+      const actions = Services.actionServiceForDoc(this.getAmpDoc());
+      actions.addToWhitelist('AMP-SIDEBAR.open');
+      actions.addToWhitelist('AMP-SIDEBAR.close');
+      actions.addToWhitelist('AMP-SIDEBAR.toggle');
+      if (this.win.MutationObserver) {
+        const sidebarObserver = new this.win.MutationObserver(() => {
+          this.storeService_.dispatch(Action.TOGGLE_SIDEBAR,
+              this.sidebar_.hasAttribute('open'));
+        });
+        sidebarObserver.observe(this.sidebar_, {attributes: open});
+      } else {
+        this.storeService_.dispatch(Action.TOGGLE_PAUSED, false);
+      } 
```

Comment:
As discussed offline, please move this whole `MutationObserver` block to the `onSidebarStateUpdate_` method, so we can observe/disconnect depending on the `sidebarState`

Reply:
Okay I implemented the fix as we discussed, and the feature works, but I'm not sure that the disconnection is working properly. When I try and make the observer a field, then the pause functionality won't work, but when I define it as a constant within the function, it does. Since the observer is defined every time it enters this function, I have doubts that it is actually disconnecting.